lib/auth/touchid/credentials.m

```diff
@@ -0,0 +1,110 @@
+// go:build touchid
+//  +build touchid
+
+// Copyright 2022 Gravitational, Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "credentials.h"
+
+#import <CoreFoundation/CoreFoundation.h>
+#import <Foundation/Foundation.h>
+#import <Security/Security.h>
+
+#include <stdlib.h>
+
+#include "common.h"
+
+BOOL matchesLabelFilter(LabelFilterKind kind, NSString *filter,
+                        NSString *label) {
+  switch (kind) {
+  case LABEL_EXACT:
+    return [label isEqualToString:filter];
+  case LABEL_PREFIX:
+    return [label hasPrefix:filter];
+  }
+  return NO;
+}
+
+int FindCredentials(LabelFilter filter, CredentialInfo **infosOut) {
```

Comment:
This should return a `size_t`.

Reply:
Done.

Reply:
Actually, we need a signed type here. OSStatus is an int32, so I think we were good to begin with. I don't expect someone will get more than an int32 worth of credentials in.

Reply:
Let's put a check in for that then? `if (count > INT_MAX) count = INT_MAX;` or something like that.

Reply:
Done.